configure.py

```diff
@@ -189,6 +191,8 @@ def configure_mode(mode):
         '-DCMAKE_C_COMPILER={}'.format(args.cc),
         '-DCMAKE_CXX_COMPILER={}'.format(args.cxx),
         '-DCMAKE_CXX_STANDARD={}'.format(args.cpp_standard),
+        '-DCMAKE_C_COMPILER_LAUNCHER={}'.format(args.ccache),
```

Comment:
as a C++ project, seastar does not use the C compiler by itself, currently, the only consumer of the `CMAKE_C_COMPILER` is the `Cooking.cmake` script generated by `cooking.sh` . this CMake script is used for building the ingredients. and `CMAKE_C_COMPILER` is currently used for building CMake based ingredients. `fmt` and `yaml-cpp` is the only two in this category. but neither of them is a C project.

so without #1649, i am afraid that we cannot benefit from this setting. but even with #1649, we still need to populate this setting down to the C projects like c-ares, dpdk, liburing and lz4, when calling `cooking_ingredient()`.

Reply:
@tchaikov - thanks for the background. I added `CMAKE_C_COMPILER_LAUNCHER` for symmetry with the C++ side of things and because this script does already set up other the CMAKE_C properties, though not for direct use of seastar as you point out.

Personally, I find that only the C++ launcher is really needed to use ccache usefully, as I'm very rarely rebuilding and of the decencies. So where does this leave us? Would you prefer I remove the C_COMPILER_LAUNCHER? I fine either way.

Reply:
@travisdowns

hi Travis,

tl;dr, i'd like to drop it.

long story: as it adds another variable which is never picked up by CMake. it's just confusing from a picky tool (CMake actually complains at seeing unused variables) or a human developer's perspective.

and after digging deeper in the dependency building machinery, i realized that `Cooking.cmake` does not rebuild the ingredients at all. because, it adds the ingredients using `ExternalProject_add()`, which does not rebuild the external projects unless `BUILD_ALWAYS` is specified, see https://cmake.org/cmake/help/latest/module/ExternalProject.html . and presumably, like you are, neither do other Seastar application developers want to hack and rebuild the dependencies once they are built. so, to avoid addressing the non-existing needs, we don't even need to populate this flag down to `Cooking.cmake` via `cooking_recipe.cmake`

Reply:
@tchaikov - sure, no problem. I've pushed an update which only sets `CMAKE_CXX_COMPILER_LAUNCHER`.

https://github.com/scylladb/seastar/commit/f4e36e91b5ef8da984cd2483296f265d354dfda6